File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        a=0
        if len(s)>len(t):
            return False
        if len(s)==0:
            return True
        for i in range (0,len(t)):
            if s[a]==t[i]:
                a+=1
            if a==len(s):
                return True
        return a==len(s)
        

File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int a=0;
        if(s.length()>t.length())return false;
        if(s.length()==0)return true;
        for(int i=0;i<t.length();i++){
            if(s.charAt(a)==t.charAt(i)){
                a++;
            }
            if(a==s.length())return true;
        }
        return a==s.length();
    }
}